examples/compute_kinematics.py

```diff
@@ -0,0 +1,439 @@
+"""
+Compute kinematics
+============================
+
+Compute displacement, velocity and acceleration data on an example dataset.
+"""
+
+# %%
+# Imports
+# -------
+
+import numpy as np
+
+# For interactive plots: install ipympl with `pip install ipympl` and uncomment
+# the following line in your notebook
+# %matplotlib widget
+# Install circle_fit in your virtual environment with `pip install circle-fit`
+from circle_fit import taubinSVD
+from circle_fit.circle_fit import convert_input
+from matplotlib import pyplot as plt
+
+import movement.analysis.kinematics as kin
+from movement import sample_data
+from movement.io import load_poses
+
+# %%
+# Load sample dataset
+# ------------------------
+# First, we load an example dataset. In this case, we select the
+# `SLEAP_three-mice_Aeon_proofread` sample data.
+file_path = sample_data.fetch_sample_data_path(
+    "SLEAP_three-mice_Aeon_proofread.analysis.h5"
+)
+
+# %%
+# Inspect the data
+# ---------------------------------
+# The data was acquired at 50 fps.
+# If we pass this info to the data loading method, the time dimension will be
+# expressed in seconds.
+# If we don't, the time dimension will be frames (starting with frame 0).
+ds = load_poses.from_sleap_file(file_path, fps=50)
+print(ds)
+
+# %%
+# We can see in the printed description of the dataset ``ds`` that it includes
+# data for three individuals(``'AEON3B_NTP'``, ``'AEON3B_TP1'``, and `
+# `'AEON3B_TP2'``), and that only one keypoint called ``centroid`` was tracked
+# in ``x`` and ``y``
+# dimensions.
+
+# %%
+# The loaded dataset ``ds`` contains two data arrays:
+# ``pose_tracks`` and ``confidence``.
+# To compute displacement, velocity and acceleration, we will need the pose
+# tracks one:
+pose_tracks = ds.pose_tracks
+
+
+# %%
+# Visualise the data
+# ---------------------------
+# First, let's visualise the trajectories of the mice in the XY plane,
+# colouring them by individual.
+
+fig, ax = plt.subplots(1, 1)
+for mouse_name, col in zip(pose_tracks.individuals.values, ["r", "g", "b"]):
+    ax.plot(
+        pose_tracks.sel(individuals=mouse_name, space="x"),
+        pose_tracks.sel(individuals=mouse_name, space="y"),
+        linestyle="-",
+        marker=".",
+        markersize=2,
+        linewidth=0.5,
+        c=col,
+        label=mouse_name,
+    )
+    ax.invert_yaxis()
+    ax.set_xlabel("x (pixels)")
+    ax.set_ylabel("y (pixels)")
+    ax.axis("equal")
+    ax.legend()
+
+# %%
+# We can see that the trajectories of the three mice are close to a circular
+# arc. Notice that the x and y axes are set to equal scales, and that the
+# origin of the coordinate system is at the top left of the image. This
+# follows the convention for SLEAP and most image processing tools.
+
+# %%
+# We can also color color the data based on its timestamp:
+fig, axes = plt.subplots(3, 1)
```

Comment:
Noted, I removed a bit the clutter with `sharey=True`.

I only set y-axis to be shared to prioritise making the x-axis and y-axis scales equal, since the trajectory is expected to be circular. Setting both `sharex=True, sharey=True` is incompatible with `ax.axis("equal")`. If I only set `sharex=True` and axis equal, it crops out the data
![image](https://github.com/neuroinformatics-unit/movement/assets/33267254/630111ba-f534-45a1-aa2d-e6b2db173d37)